azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -372,10 +417,177 @@ private String getDefaultValue(String defaultValue, String fallBack, String patt
     }
 
     private boolean isJarProject() {
-        return getProject().getPackaging().equalsIgnoreCase("jar");
+        return Utils.isJarPackagingProject(project.getPackaging());
     }
 
     private WebAppConfiguration getWebAppConfigurationWithoutValidation() throws AzureExecutionException {
         return new V2NoValidationConfigurationParser(this, new V2ConfigurationValidator(this)).getWebAppConfiguration();
     }
+
+    private WebAppConfiguration chooseExistingWebappForConfiguration()
+            throws AzureExecutionException, AzureAuthFailureException {
+        final Azure az = getAzureClientByAuthType();
+        final TextIO textIO = TextIoFactory.getTextIO();
+        final Subscription[] subscriptions = az.subscriptions().list().toArray(new Subscription[0]);
+        final Subscription targetSubscription;
+        if (subscriptions.length == 0) {
+            throw new AzureExecutionException("Cannot find any subscriptions in current account.");
+        } else if (subscriptions.length == 1) {
+            Log.info(String.format("There is only one subscription '%s' in your account, will use it automatically.",
+                    TextUtils.blue(SubscriptionOption.getSubscriptionName(subscriptions[0]))));
+            targetSubscription = subscriptions[0];
+        } else {
+            final String defaultId = az.getCurrentSubscription() != null ? az.getCurrentSubscription().subscriptionId()
+                    : null;
+
+            final List<SubscriptionOption> wrapSubs = Arrays.stream(subscriptions).map(t -> new SubscriptionOption(t))
+                    .sorted()
```

Comment:
is `SubscriptionOption` comparable?
as far as I know it's meaningless to sort a stream of un-comparable objects